marc_to_solr/lib/alma_traject_config.rb

```diff
@@ -0,0 +1,1339 @@
+
+# Supplement to:
+#    772 XX at
+to_field 'supplement_to_display', extract_marc('772at', trim_punctuation: true)
+
+# Linking notes:
+#    580 XX a
+to_field 'linking_notes_display', extract_marc('580a')
+
+# Subseries of:
+#    760 XX at
+to_field 'subseries_of_display', extract_marc('760at', trim_punctuation: true)
+
+# Has subseries:
+#    762 XX at
+to_field 'has_subseries_display', extract_marc('762at', trim_punctuation: true)
+
+# Series:
+#    400 XX abcdefgklnpqtuvx
+#    410 XX abcdefgklnptuvx
+#    411 XX acdefgklnpqtuv
+#    440 XX anpvx
+#    490 XX avx
+#    800 XX abcdefghklmnopqrstuv
+#    810 XX abcdefgklnt
+#    811 XX abcdefghklnpqstuv
+#    830 XX adfghklmnoprstv
+#    840 XX anpv
+# only includes 490 if the value is different from the other fields
+to_field 'series_display', extract_marc('400abcdefgklnpqtuvx:410abcdefgklnptuvx:411acdefgklnpqtuv:440anpvx:800abcdefghklmnopqrstuv:810abcdefgklnt:811abcdefghklnpqstuv:830adfghklmnoprstv:840anpv') do |record, accumulator|
+  without_punct = accumulator.map { |f| Traject::Macros::Marc21.trim_punctuation(f) }
+  MarcExtractor.cached('490avx').collect_matching_lines(record) do |field, spec, extractor|
+    series = extractor.collect_subfields(field, spec).first
+    accumulator << series unless without_punct.include?(Traject::Macros::Marc21.trim_punctuation(series))
+  end
+end
+
+# a subset of the series fields and subfields to link to "More in this series"
+to_field 'more_in_this_series_t' do |record, accumulator|
+  MarcExtractor.cached('440anp:830anp').collect_matching_lines(record) do |field, spec, extractor|
+    str = extractor.collect_subfields(field, spec).first
+    if str
+      str = str.slice(field.indicator2.to_i, str.length)
+      if str.nil? || str.empty?
+        logger.error "#{record['001']} - Non-filing characters >= title length"
+      else
+        accumulator << str
+      end
+    else
+      logger.error "#{record['001']} - Missing 440/830 $a"
+    end
+  end
+  accumulator << everything_through_t(record, '800:810:811')
+  accumulator.flatten!.map! { |f| Traject::Macros::Marc21.trim_punctuation(f) }
+end
+
+
+# Other version(s):
+#    3500 020Z020A
+#    3500 020A020Z
+#    3500 024A022A
+#    3500 022A024A
+#    3500 BBID776W
+#    3500 BBID787W
+#    3500 776X022A
+#    3500 022A776X
+#    3500 020A776Z
+#    3500 776Z020A
+# to_field 'Other version(s)_display', extract_marc()
+# # #    3500 020Z020A
+# # #    3500 020A020Z
+# # #    3500 024A022A
+# # #    3500 022A024A
+# # #    3500 BBID776W
+# # #    3500 BBID787W
+# # #    3500 776X022A
+# # #    3500 022A776X
+# # #    3500 020A776Z
+# # #    3500 776Z020A
+
+to_field 'geo_related_record_display', extract_marc('772at:7733abdghikmnoprst:777at', trim_punctuation: true)
+
+# Contained in:
+#    3500 BBID773W
+# it will not change but we need to do more work to retrieve info from the link record.
+# if there is a 773 we need to retrieve the holding and item information from that linked record.
```

Comment:
We have an issue for that https://github.com/pulibrary/marc_liberation/issues/800.